File: packages/urdu-js/src/string-utils/extract.ts
export function extractUrdu(text: string): string[] {
  const segments: string[] = []
  let current = ''
  for (const ch of text) {
    const cp = ch.codePointAt(0) ?? 0
    const isUrdu =
      (cp >= 0x0600 && cp <= 0x06ff) ||
      (cp >= 0x0750 && cp <= 0x077f) ||
      (cp >= 0xfb50 && cp <= 0xfdff) ||
      (cp >= 0xfe70 && cp <= 0xfeff)
    if (isUrdu) {
      current += ch
    } else if (ch === ' ' && current) {
      current += ch
    } else {
      const t = current.trim()
      if (t) segments.push(t)
      current = ''
    }
  }
  const t = current.trim()
  if (t) segments.push(t)
  return segments
}

File: packages/urdu-dart/lib/src/string_utils/extract.dart
/// Extracts continuous segments of Urdu text.
List<String> extractUrdu(String text) {
  final segments = <String>[];
  final buf = StringBuffer();
  for (final r in text.runes) {
    final isUrdu = (r >= 0x0600 && r <= 0x06ff) ||
        (r >= 0x0750 && r <= 0x077f) ||
        (r >= 0xfb50 && r <= 0xfdff) ||
        (r >= 0xfe70 && r <= 0xfeff);
    if (isUrdu) {
      buf.writeCharCode(r);
    } else if (r == 0x0020 && buf.isNotEmpty) {
      buf.writeCharCode(r);
    } else {
      final t = buf.toString().trim();
      if (t.isNotEmpty) segments.add(t);
      buf.clear();
    }
  }
  final t = buf.toString().trim();
  if (t.isNotEmpty) segments.add(t);
  return segments;
}
